c++/c++-quiz.md

```diff
@@ -0,0 +1,272 @@
+- []t median(const my_arrary &a);
+- []re>a) The argument is passed as a reference so the function receives a copy that can be modified without affecting the original value.
+- [x] The argument is passed as a reference,so all if the passed my_array object is large, the program will require less time and memory.
+- [] Actually object can't be passed as regular variable because they require a constructor call. Therefore a const reference is the only way to pass class instances to functions.
+- [] There are no benefits because a reference and an object are treated as the same thing.
+</pre>
+#### Q7. What's the storage occupied by u1?
+<pre>union {
+unit16_t a;
+unit32_t b;
+int8_t c;
+} u1;
+- [x] 4 bytes You can see example [here](https://en.cppreference.com/w/cpp/language/union)
+- [] 7 bytes
+- [] 8 bytes
+- [] 2 bytes
+</pre>
+#### Q8. Which of the following operators is overloadable?
+
+    - [] ?:
+    - [x] new
+    - [] ::
+    - [] .
+
+#### Q9. Which of the following shows the contents of vector pointed by v1 and v2 after running this code?
+<pre>std:: vector<int> *v1 = new std:: vector<int> ({1,2,3});
+std:: vector<int> *v2;
+v2=v1;
+v1->push_back(4);
+v2->push_back(5);
+
+- [] *v1:{1,2,3,4};
+   *v2:{5};
+- [] *v1:{1,2,3,4'5};
+   *v2:{1,2,3,4,5};
+- [x] Error
+- [] *v1:{1,2,3,4};
+   *v2:{1,2,3,5};
+</pre>
+
+#### Q10. Which of the following is not a difference between a class and a struct?
+- []re>a) Because structor are part of the c programming language there some complexity between c and c++ struct This is not the case with classes.
+- [] Classes may have member function;struct private.
+- [x] The default acces specifier for members of struct is public, where as for member of class , it is private. -> You can see the answer [here](https://www.fluentcpp.com/2017/06/13/the-real-difference-between-struct-class/)
+- [] Template type parameters can be declared with , but not with the struct keyword.
+</pre>
+#### Q11. Suppose you need to keep a data struct with permission to access some resource base on the days of week , but you can't be use a bool variable for each day.you need to use one bit per day of the week.
+which of the following is a correct implementation of a structure with bit fields for this application?
+
+    - [] typedef struct {
+    		int sunday:1;
+    		int monday:1;
+    		// more days
+    		int friday:1;
+    		int satarday:1;
+    		- [x]ekdays;  That syntax says that each variable size is 1 bit. 'bit' is not a type in C++.
+    - [] typedef char[7]: weekdays;
+    - [] typedef struct {
+    		bit sunday:1;
+    		bit monday:1;
+    		// more days
+    		bit fridyas:1;
+    		bit satarday:1;
+    		} weekdays;
+    - [] typedef struct {
+    		bit sunday;
+    		bit monday;
+    		// more days
+    		bit friday;
+    		bit satarday;
+    		} weekdays;
+
+#### Q12. What is an lvalue?
+- []re>a) It's a constant expression , meaning an expression composed of constants and operations.
```

Comment:
could you please take a look maybe there is something wrong? and we should remove `re>a)`
```
- []re>a) It
```